src/node_lib.h

```diff
@@ -0,0 +1,457 @@
+#ifndef SRC_NODE_LIB_H_
+#define SRC_NODE_LIB_H_
+
+#include <string>
+#include <vector>
+#include <map>
+#include <functional>
+#include <initializer_list>
+#include "v8.h"
+#include "uv.h"
+#include "node.h"
+
+namespace node {
+
+namespace internal {  // internals, provided for experienced users
+
+/**
+ * @brief Returns the `v8::Isolate` for Node.js.
+ *
+ * Returns a pointer to the currently used `v8::Isolate`, if the Node.js engine
+ * is initialized already.
+ * *Important* Use with caution, changing this object might break Node.js.
+ * @return Pointer to the `v8::Isolate`.
+ */
+v8::Isolate* isolate();
+
+/**
+ * @brief Returns the `node::Environment` for Node.js.
+ *
+ * Returns a pointer to the currently used `node::Environment`, if the Node.js
+ * engine is initialized already.
+ * *Important* Use with caution, changing this object might break Node.js.
+ * @return Pointer to the `node::Environment`.
+ */
+Environment* environment();
```

Comment:
I’m sorry I’ve not really been keeping up with this, but at its core the issue remains – these APIs lock Node.js *more* into maintaining global state where it shouldn’t, and that’s not a good thing.

In particular, it’s just not going to work with multi-threaded embedders, the way it is implemented right now. But even without that, conceptually `Environment*` instances are not global singletons.